stan/math/rev/constraint/sum_to_zero_constrain.hpp

```diff
@@ -15,6 +15,35 @@
 namespace stan {
 namespace math {
 
+namespace internal {
+
+/**
+ * The reverse pass backprop for the sum_to_zero_constrain on
+ * vectors. This is separated out so it can also be called by
+ * simplex_constrain.
+ */
+template <typename T>
+void sum_to_zero_vector_backprop(T&& y_adj, const Eigen::VectorXd& z_adj) {
+  const auto N = y_adj.size();
```

Comment:
I'd leave a small comment so people know `z_adj` is `y_adj.size() + 1`

Reply:
I added `@param`s to the docstring and put the sizes there